packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -165,27 +200,21 @@ def _get_tx_data(
                 "send_raw_transaction unsuccessful! Insufficient funds."
             )
 
+        # Store gas price if we have successfully detected the strategy and we receive an underpriced status,
+        # so that the repricing mechanism is triggered next time.
+        if rpc_status == RPCResponseStatus.UNDERPRICED and gas_price:
+            self.params.gas_price = gas_price
```

Comment:
Previously we stored it all the time, why the change now?

Reply:
I think that is not 100% correct. The [current code](https://github.com/valory-xyz/open-autonomy/blob/main/packages/valory/skills/transaction_settlement_abci/behaviours.py#L200) that sets gas_price ATM won't be executed due to what I think it is a [bug](https://github.com/valory-xyz/open-autonomy/blob/main/packages/valory/skills/transaction_settlement_abci/behaviours.py#L168-L172): we cannot make because we immediately return if RPCResponseStatus is not SUCCESS

```
        if rpc_status != RPCResponseStatus.SUCCESS:
            self.context.logger.warning(
                f"send_raw_transaction unsuccessful! Received: {rpc_status}"
            )
            return tx_data
```

Reply:
I am not sure about this PR. The decision to return when we implemented the tx settlement skill was because a "replacement transaction underpriced" error means that we have not repriced correctly, which cannot happen since the ledger will take care of the correct increase. If we received a non-successful response, then we thought it would be better to use another keeper (therefore return immediately).

To clarify, we only replace txs that have been sent. If a tx does not have a nonce etc., then we cannot replace a non-existing tx. The error that has been mentioned, "Transaction underpriced" is different and needs further investigation on what this error means, as I believe it is not the same.